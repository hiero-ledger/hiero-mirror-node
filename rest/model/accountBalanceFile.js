// SPDX-License-Identifier: Apache-2.0

class AccountBalanceFile {
  static tableName = 'account_balance_file';
  static CONSENSUS_TIMESTAMP = 'consensus_timestamp';
}

export default AccountBalanceFile;
