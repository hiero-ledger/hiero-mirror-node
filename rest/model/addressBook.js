// SPDX-License-Identifier: Apache-2.0

class AddressBook {
  /**
   * Parses address_book table columns into object
   */
  constructor(addressBook) {
    // explicitly assign properties to restict properties and allow for composition in other models
    this.endConsensusTimestamp = addressBook.end_consensus_timestamp;
    this.fileData = addressBook.file_data;
    this.fileId = addressBook.file_id;
    this.nodeCount = addressBook.node_count;
    this.startConsensusTimestamp = addressBook.start_consensus_timestamp;
  }

  static tableAlias = 'adb';
  static tableName = 'address_book';

  static END_CONSENSUS_TIMESTAMP = 'end_consensus_timestamp';
  static FILE_DATA = 'file_data';
  static NODE_COUNT = 'node_count';
  static FILE_ID = 'file_id';
  static START_CONSENSUS_TIMESTAMP = 'start_consensus_timestamp';

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default AddressBook;
