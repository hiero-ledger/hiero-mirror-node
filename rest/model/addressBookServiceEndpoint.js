// SPDX-License-Identifier: Apache-2.0

class AddressBookServiceEndpoint {
  /**
   * Parses crypto_allowance table columns into object
   */
  constructor(serviceEndpoint) {
    // explicitly assign properties to restict properties and allow for composition in other models
    this.consensusTimestamp = serviceEndpoint.consensus_timestamp;
    this.domainName = serviceEndpoint.domain_name;
    this.ipAddressV4 = serviceEndpoint.ip_address_v4;
    this.nodeId = serviceEndpoint.node_id;
    this.port = serviceEndpoint.port;
  }

  static tableAlias = 'abse';
  static tableName = 'address_book_service_endpoint';

  static CONSENSUS_TIMESTAMP = 'consensus_timestamp';
  static DOMAIN_NAME = 'domain_name';
  static IP_ADDRESS_V4 = 'ip_address_v4';
  static NODE_ID = 'node_id';
  static PORT = 'port';

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default AddressBookServiceEndpoint;
