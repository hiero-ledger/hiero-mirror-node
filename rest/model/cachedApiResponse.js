// SPDX-License-Identifier: Apache-2.0

/**
 * API response cached in Redis.
 */
class CachedApiResponse {
  constructor(statusCode, headers, body, compressed) {
    this.statusCode = statusCode;
    this.body = body;
    this.headers = headers;
    this.compressed = compressed;
  }
}

export default CachedApiResponse;
