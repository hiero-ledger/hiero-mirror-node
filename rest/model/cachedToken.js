// SPDX-License-Identifier: Apache-2.0

/**
 * Cached token object to store token's decimals, freeze status and kyc status
 */
class CachedToken {
  constructor(token) {
    this.decimals = token.decimals;
    this.freezeStatus = token.freeze_status;
    this.kycStatus = token.kyc_status;
  }
}

export default CachedToken;
