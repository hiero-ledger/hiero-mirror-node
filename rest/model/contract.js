// SPDX-License-Identifier: Apache-2.0

import _ from 'lodash';

class Contract {
  static tableAlias = 'c';
  static tableName = 'contract';

  static FILE_ID = 'file_id';
  static ID = 'id';
  static INITCODE = 'initcode';
  static RUNTIME_BYTECODE = 'runtime_bytecode';

  /**
   * Parses contract table columns into object
   */
  constructor(contract) {
    Object.assign(
      this,
      _.mapKeys(contract, (v, k) => _.camelCase(k))
    );
  }

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default Contract;
