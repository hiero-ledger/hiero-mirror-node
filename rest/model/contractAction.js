// SPDX-License-Identifier: Apache-2.0

import _ from 'lodash';

class ContractAction {
  static tableAlias = 'cact';
  static tableName = 'contract_action';

  static CALL_DEPTH = 'call_depth';
  static CALL_OPERATION_TYPE = 'call_operation_type';
  static CALL_TYPE = 'call_type';
  static CALLER = 'caller';
  static CALLER_TYPE = 'caller_type';
  static CONSENSUS_TIMESTAMP = 'consensus_timestamp';
  static GAS = 'gas';
  static GAS_USED = 'gas_used';
  static INDEX = 'index';
  static INPUT = 'input';
  static PAYER_ACCOUNT_ID = 'payer_account_id';
  static RECIPIENT_ACCOUNT = 'recipient_account';
  static RECIPIENT_ADDRESS = 'recipient_address';
  static RECIPIENT_CONTRACT = 'recipient_contract';
  static RESULT_DATA = 'result_data';
  static RESULT_DATA_TYPE = 'result_data_type';
  static VALUE = 'value';

  /**
   * Parses contract_action table columns into object
   */
  constructor(contractAction) {
    Object.assign(
      this,
      _.mapKeys(contractAction, (v, k) => _.camelCase(k))
    );
  }

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default ContractAction;
