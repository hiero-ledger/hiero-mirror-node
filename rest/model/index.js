// SPDX-License-Identifier: Apache-2.0

import AccountBalanceFile from './accountBalanceFile';
import AddressBook from './addressBook';
import AddressBookEntry from './addressBookEntry';
import AddressBookServiceEndpoint from './addressBookServiceEndpoint';
import AssessedCustomFee from './assessedCustomFee';
import CachedApiResponse from './cachedApiResponse';
import CachedToken from './cachedToken';
import Contract from './contract';
import CryptoAllowance from './cryptoAllowance';
import ContractAction from './contractAction';
import ContractLog from './contractLog';
import ContractResult from './contractResult';
import ContractStateChange from './contractStateChange';
import ContractTransactionHash from './contractTransactionHash';
import CryptoTransfer from './cryptoTransfer';
import CustomFee from './customFee';
import CustomFeeLimits from './customFeeLimits';
import Entity from './entity';
import EthereumTransaction from './ethereumTransaction';
import ExchangeRate from './exchangeRate';
import FileData from './fileData';
import FixedFee from './fixedFee';
import Node from './node';
import NetworkNode from './networkNode';
import Nft from './nft';
import NftHistory from './nftHistory';
import NftTransfer from './nftTransfer';
import NodeStake from './nodeStake';
import RecordFile from './recordFile';
import ServiceEndpoint from './serviceEndpoint';
import SignatureType from './signatureType';
import StakingRewardTransfer from './stakingRewardTransfer';
import Token from './token';
import TokenAllowance from './tokenAllowance';
import TokenFreezeStatus from './tokenFreezeStatus';
import TokenKycStatus from './tokenKycStatus';
import TokenAccount from './tokenAccount';
import TokenBalance from './tokenBalance';
import TokenTransfer from './tokenTransfer';
import TokenType from './tokenType';
import TopicMessage from './topicMessage';
import TopicMessageLookup from './topicMessageLookup';
import Transaction from './transaction';
import TransactionHash from './transactionHash';
import TransactionId from './transactionId';
import TransactionResult from './transactionResult';
import TransactionType from './transactionType';
import FeeSchedule from './feeSchedule';
import ContractState from './contractState';

export {
  AccountBalanceFile,
  AddressBook,
  AddressBookEntry,
  AddressBookServiceEndpoint,
  AssessedCustomFee,
  CachedApiResponse,
  CachedToken,
  Contract,
  CryptoAllowance,
  ContractAction,
  ContractLog,
  ContractResult,
  ContractStateChange,
  ContractTransactionHash,
  CryptoTransfer,
  CustomFee,
  CustomFeeLimits,
  Entity,
  EthereumTransaction,
  ExchangeRate,
  FileData,
  FixedFee,
  Node,
  NetworkNode,
  Nft,
  NftHistory,
  NftTransfer,
  NodeStake,
  RecordFile,
  ServiceEndpoint,
  SignatureType,
  StakingRewardTransfer,
  Token,
  TokenAllowance,
  TokenFreezeStatus,
  TokenKycStatus,
  TokenAccount,
  TokenBalance,
  TokenTransfer,
  TokenType,
  TopicMessage,
  TopicMessageLookup,
  Transaction,
  TransactionHash,
  TransactionId,
  TransactionResult,
  TransactionType,
  FeeSchedule,
  ContractState,
};
