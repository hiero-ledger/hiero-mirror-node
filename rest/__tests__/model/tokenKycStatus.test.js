// SPDX-License-Identifier: Apache-2.0

import {TokenKycStatus} from '../../model';

describe('TokenKycStatus', () => {
  describe('invalid id', () => {
    [-1, 3].forEach((value) =>
      test(`${value}`, () => {
        expect(() => new TokenKycStatus(value)).toThrowErrorMatchingSnapshot();
      })
    );
  });

  test('toJSON', () => {
    const input = [new TokenKycStatus(0), new TokenKycStatus(1), new TokenKycStatus(2)];
    expect(JSON.stringify(input)).toEqual(JSON.stringify(['NOT_APPLICABLE', 'GRANTED', 'REVOKED']));
  });
});
