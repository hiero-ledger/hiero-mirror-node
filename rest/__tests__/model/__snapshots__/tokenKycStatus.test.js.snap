// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TokenKycStatus invalid id -1 1`] = `"Invalid token kyc status id -1"`;

exports[`TokenKycStatus invalid id 3 1`] = `"Invalid token kyc status id 3"`;
